### DISCUSSION TITLE
Hard to add new archives to the DB on Windows

### DISCUSSION THREAD

**Discussion by soarpasser** [Difegue/LANraragi#700] [Q&A] [answered]
0.8.6 Windows

Currently in LR, to "properly" add a new archive to the database (i.e. adding it to a category, adding tags etc.) I need to go through two major steps:

1. Put the archive into the content folder and hitting the "Rescan archive directory" button (otherwise LR will simply not recognize this new archive as per my testing).
![image](https://user-images.githubusercontent.com/74951691/193251583-9ff7f4d1-6d41-4028-891f-d4a3634ff628.png)
Problem with this:
-Rescanning the directory takes quite an amount of time (exceptionally so when LR loads up for the first time) when you have many archives in the database, which in itself is already quite annoying. But it gets worse:

2. Entering the new archive's file name in the search bar and applying a search.
![image](https://user-images.githubusercontent.com/74951691/193254011-716ad3da-7ba0-4b40-b707-fb81af3d43a0.png)
This in itself opens up a whole can of worms, because the search bar simply refuses to work most of the time. See #657 for details.
In other cases the whole thing can also get stuck in infinite loading mode like this:
![image](https://user-images.githubusercontent.com/74951691/193254996-ce89adc0-b03c-4be9-adfa-6ee38b228a7a.png)

Only after going through these two excruciating steps (which depending on the situation can take minutes) can you actually get to see the new archive and edit it. This is very much not enjoyable when you're adding multiple archives to the database while reading along.

Now, some may ask "But why not use the 'Add Archives' tab built into LR and just drag&drop?

Two reasons. Mainly, the "add archives" thing doesn't function very well when you're adding many archives or archives with a large file size - I attempted to add around 6 archives that ranges from ~200MB to 600MB size into the database with the "add archives" function, and LR got stuck processing that task for an hour - while a copy&paste in the windows explorer takes less than 30 seconds.

Then there's the fact that "Add archives" drops the added archives straight into the root content folder directory. Personally I prefer having the content folder contain several different child folders by category in the windows explorer so I can view&manage them more easily when I check the explorer locally. LR just sticks everything I add into the root folder, which can be bothersome in this situation.

Basically, trying to add a new archive to the database currently is quite a pain in the arse, at least for me. If there are any optimizations that can be done in the future to speed up this process or make it less painful, it would be greatly appreciated.

**Comment by Difegue** [comment, marked as answer]
👋 Converted this to a discussion as it's more of an external issue with the windows port than a fixable bug;

Have you tried simply restarting the server instead of triggering a full rescan? 
I know the filewatcher can be busted on windows depending on a ton of external factors I don't have control over, but after a server restart it usually works long enough to figure out what files were added.

**Reply by soarpasser** [reply]
This does in fact seem to work better - I've actually been trying this exact suggestion since a few days ago - i.e. simply pressing "stop" in the taskbar context menu and "start" again, then triggering the search; such a method would usually get the search thing to pick up newly added archives much quicker, though I haven't yet checked if it helps with the accuracy of existing archives.

Some insights on how Windows breaks the filewatcher would be appreciated as well!

**Reply by Difegue** [reply]
I wish I knew!  
I'm guessing if using WSL1, that some of the inotify instructions aren't translated correctly to the NT kernel. WSL2 supposedly works better for this, but the extra virtualization layer means that it is going to be noticeably slower regarding filesystem access.
